File: FangZhiHu/app/src/main/java/network/exception/SendException.java
package network.exception;

/**
 * Created by yuantongqin on 2017/8/15.
 */

public class SendException extends Exception {

    private int code;
    private String msg;
    private String data;
    public int customState;

    public SendException(Throwable throwable, int code) {
        super(throwable);
    }

    public SendException(Throwable throwable, int code, String msg) {
        super(throwable);
        this.code = code;
        this.msg = msg;
    }
    public SendException(Throwable throwable,int customState, int code, String msg, String data) {
        super(throwable);
        this.customState = customState;
        this.code = code;
        this.msg = msg;
        this.data = data;
    }

    public SendException(Throwable throwable, int code, String msg, String data) {
        super(throwable);
        this.code = code;
        this.msg = msg;
        this.data = data;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public String getData() {
        return data;
    }


}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/excepition/SendException.kt
package network.excepition


/**
 * Created by tongqinyuan on 2018/3/7.
 * 这个是用来传递消息的错误类
 */
class SendException : Exception {

    var code:Int = 0
    var msg:String? = null
    var data:String? = null
        private set
    var customState:Int = 0


    constructor(throwable:Throwable , code:Int ) :super(throwable){

    }


    constructor(throwable:Throwable , code:Int , msg:String ):super(throwable){
        this.code = code
        this.msg = msg
    }

    constructor(throwable:Throwable , code:Int , msg:String, data:String ):super(throwable){
        this.code = code
        this.msg = msg
        this.data = data
    }

    constructor(throwable:Throwable , code:Int , msg:String,data:String,customState:Int ):super(throwable){
        this.customState = customState;
        this.code = code;
        this.msg = msg;
        this.data = data;
    }






}